osidb/api_views.py

```diff
@@ -811,6 +819,49 @@ def destroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         return super().destroy(request, *args, **kwargs)
 
 
+class FlawAvailableView(
+    RudimentaryUserPathLoggingMixin,
+    APIView,
+):
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    http_method_names = ["get"]
+
+    def get(self, request, *args, **kwargs):
+        """
+        Report whether a flaw is available based on the following criteria:
+
+        1) Public / work on flaw is done: 200 status (yes)
+        2) Not public and work not done / flaw does not exist: 404 status (no)
+        """
+
+        pk = self.kwargs["flaw_id"]
```

Comment:
Should probably be named `cve_id` and not `flaw_id`, the latter and the variable name strongly imply that this is Flaw's primary key which is a UUID.